tests/expected/python/class_pybind.cpp

```diff
@@ -91,6 +91,12 @@ PYBIND11_MODULE(class_py, m_) {
         .def(py::init<const int&>(), py::arg("arg"))
         .def(py::init<const double&>(), py::arg("arg"));
 
+    py::class_<FastSet, std::shared_ptr<FastSet>>(m_, "FastSet")
+        .def(py::init<>())
+        .def("__len__",[](FastSet* self){return self->size();})
```

Comment:
Just checking, are there non-member STL versions of these functions?

Eg std::distance(std::end(self), std::begin(self)) ?

Not sure if it ever makes a difference but might take more advantage of the STL?

Reply:
I don't understand the question. The assumption is that the containers with dunder methods are either subclassed from STL containers or provide an efficient implementation of the method. I don't understand how `std::difference` would take advantage of the STL, especially if the container doesn't use the STL.

Reply:
Just thinking that it might be better to use non-member-function versions of the `size()` and `find()` functions since I would expect the std:: namespace versions to be more general, i.e. `std::distance` from <iterator> and `std::find` from <algorithm>.  Both of these would only need `std::begin(obj)` and `std::end(obj)` to be implemented.  I guess this is like the difference between using `obj.begin()` vs `std::begin(obj)`.  Although I suppose std::find would be significantly less efficient for many data structures.

Thoughts?

P.S. requested changes was to request comment on this Q.

Reply:
Yeah I guess that makes sense but it would need the annotated class to define a `begin()` and an `end()` method, which I believe is easier to do. I'll update.